### PR TITLE
logging: Fix case where logging strings were not stripped from the binary when dictionary logging was used

### PR DESCRIPTION
**Pull request by nordic-krch**
Due to `LOG_OUTPUT` beings selected by any logging backend, logging strings were not stripped from the binary. It can be seen in `samples/subsys/logging/dictionary` where it is expected that strings are stripped by they weren't.

### CODE REVIEW

**Review by karstenkoenig — DISMISSED** [review]

**Review by dcpleung — DISMISSED** [review]

**Review by cfriedt — DISMISSED** [review]

File: include/zephyr/logging/log_output.h [hunk 1]
```diff
@@ -190,11 +190,35 @@ void log_output_msg_syst_process(const struct log_output *log_output,
...
 /** @brief Flush output buffer.
  *
  * @param output Pointer to the log output instance.
  */
-void log_output_flush(const struct log_output *output);
+static inline void log_output_flush(const struct log_output *output)
+{
+	log_output_write(output->func, output->buf,
+		     output->control_block->offset,
+		     output->control_block->ctx);
```
Comment (pdgendt) [inline include/zephyr/logging/log_output.h:219]: The formatting comment is valid here, IMO.

File: subsys/logging/log_output.c [hunk 2]
```diff
@@ -785,10 +763,10 @@ void log_output_dropped_process(const struct log_output *output, uint32_t cnt)
...
 	len = snprintk(buf, sizeof(buf), "%d", cnt);
 
-	buffer_write(outf, (uint8_t *)prefix, sizeof(prefix) - 1,
+	log_output_write(outf, (uint8_t *)prefix, sizeof(prefix) - 1,
 		     output->control_block->ctx);
-	buffer_write(outf, buf, len, output->control_block->ctx);
-	buffer_write(outf, (uint8_t *)postfix, sizeof(postfix) - 1,
+	log_output_write(outf, buf, len, output->control_block->ctx);
+	log_output_write(outf, (uint8_t *)postfix, sizeof(postfix) - 1,
 		     output->control_block->ctx);
```
Comment (pdgendt) [inline subsys/logging/log_output.c:770]: Same here

File: include/zephyr/logging/log_output.h (hunk 1 above)
Comment (nordic-krch) [reply include/zephyr/logging/log_output.h]: run clang-format on that commit.

**Comment by nordic-krch** [comment]
@dcpleung @karstenkoenig @cfriedt can you reapprove? Only clang-format changes.

**Review by karstenkoenig — APPROVED** [review]

**Review by pdgendt — APPROVED** [review]

**Review by cfriedt — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Pull request merged**